client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java

```diff
@@ -423,6 +424,21 @@ public void testAssumeLiteralUnderscoreInMetadataCallsForNonConformingClients()
         assertThat(parameters.isAssumeLiteralNamesInMetadataCallsForNonConformingClients()).isFalse();
     }
 
+    @Test
+    public void testTimezone()
+            throws SQLException
+    {
+        TrinoDriverUri defaultParameters = createDriverUri("jdbc:trino://localhost:8080");
+        assertThat(defaultParameters.getTimeZone()).isEqualTo(ZoneId.systemDefault());
+
+        TrinoDriverUri parameters = createDriverUri("jdbc:trino://localhost:8080?timezone=Asia/Kolkata");
+        assertThat(parameters.getTimeZone()).isEqualTo(ZoneId.of("Asia/Kolkata"));
+
+        assertThatThrownBy(() -> createDriverUri("jdbc:trino://localhost:8080?timezone=Asia/NOT_FOUND"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageStartingWith("Specified timezone is not supported: Asia/NOT_FOUND");
```

Comment:
I couldn't use `assertInvalid` here because it expects a `SqlException` to be thrown.

I cannot throw `SqlException` (or any checked exception) from `TrinoUri#getTimeZone` because the callers don't expect it.

It looks like the properties which throw `SqlException` are handled in `TrinoDriverUri` instead of `TrinoUri`.

So should I move timezone to `TrinoDriverUri` instead of `TrinoUri`? But then what about the CLI? Or is there some other recommendation on how to handle exceptions in `TrinoUri` class.